feat(server side): written data minification class

Smipe-a opened the pull request:
Was written data minification class.

Smipe-a committed:
fix(server side): fixed bug appearance tag

Was fixed bug appearance tag.

Smipe-a committed:
feat(server side): created class exception throw SAXExeception

Was created class exception throw SAXExeception.

Smipe-a committed:
fix(server side): add throw in class OSMMinifier and rewritte exception

Was added throw in class OSMMinifier and rewritte exception InvalidOSMDataException.

Smipe-a committed:
feat(server side): written test class OSMMinifier

Was written test class OSMMinifier.

VladimirBalun commented on ServerSide/resourceserver/src/main/java/ru.servers.resourceserver/osmexception/InvalidOSMDataException.java:
```diff
@@ -0,0 +1,11 @@
+
+public class InvalidOSMDataException extends SAXException {
+
+    public InvalidOSMDataException(String message, Exception e) {
```
Unimportant 'e' parameter in the method. Yopu need to rmove it.

VladimirBalun commented on ServerSide/resourceserver/src/test/java/ru/servers/resourceserver/OSMMinifierTest.java:
```diff
@@ -0,0 +1,107 @@
+
+    private OSMMinifier osmMinifier = new OSMMinifier();
+
+    @Test // deleting attributes in tag node
```
Unimportant comment. You need to remove it.

VladimirBalun commented on ServerSide/resourceserver/src/test/java/ru/servers/resourceserver/OSMMinifierTest.java:
```diff
@@ -0,0 +1,107 @@
+
+    @Test // deleting attributes in tag node
+    public void checkingNodeMinification() throws InvalidOSMDataException {
+
```
Extra empty line.

VladimirBalun commented on ServerSide/resourceserver/src/test/java/ru/servers/resourceserver/OSMMinifierTest.java:
```diff
@@ -0,0 +1,107 @@
+        Assert.assertEquals(minifiedData, resultMinification);
+    }
+
+    @Test // deleting attributes in tag way
```
Unimportant comment. You need to remove it.

VladimirBalun commented on ServerSide/resourceserver/src/test/java/ru/servers/resourceserver/OSMMinifierTest.java:
```diff
@@ -0,0 +1,107 @@
+
+    @Test // deleting attributes in tag way
+    public void checkingWayMinification() throws InvalidOSMDataException {
+
```
Extra empty line.

VladimirBalun commented on ServerSide/resourceserver/src/test/java/ru/servers/resourceserver/OSMMinifierTest.java:
```diff
@@ -0,0 +1,107 @@
+        Assert.assertEquals(minifiedData, resultMinification);
+    }
+
+    @Test // deleting attributes in tag relation
```
Unimportant comment. You need to remove it.

VladimirBalun commented on ServerSide/resourceserver/src/test/java/ru/servers/resourceserver/OSMMinifierTest.java:
```diff
@@ -0,0 +1,107 @@
+
+    @Test // deleting attributes in tag relation
+    public void checkingRelationMinification() throws InvalidOSMDataException {
+
```
Extra empty line.

VladimirBalun commented on ServerSide/resourceserver/src/test/java/ru/servers/resourceserver/OSMMinifierTest.java:
```diff
@@ -0,0 +1,107 @@
+
+    @Test(expected = InvalidOSMDataException.class) // minification invalid string osmData
+    public void checkingInvalidStringMinification() throws InvalidOSMDataException {
+
```
Extra empty line.

VladimirBalun commented on ServerSide/resourceserver/src/test/java/ru/servers/resourceserver/OSMMinifierTest.java:
```diff
@@ -0,0 +1,107 @@
+
+    @Test(expected = InvalidOSMDataException.class) // minification empty string osmData
+    public void checkingEmptyStringMinification() throws InvalidOSMDataException {
+
```
Extra empty line.

VladimirBalun commented on ServerSide/resourceserver/src/test/java/ru/servers/resourceserver/OSMMinifierTest.java:
```diff
@@ -0,0 +1,107 @@
+        Assert.assertEquals(minifiedData, resultMinification);
+    }
+
+    @Test(expected = InvalidOSMDataException.class) // minification invalid string osmData
```
Unimportant comment. You need to remove it.

VladimirBalun commented on ServerSide/resourceserver/src/test/java/ru/servers/resourceserver/OSMMinifierTest.java:
```diff
@@ -0,0 +1,107 @@
+        Assert.assertEquals(minifiedData, resultMinification);
+    }
+
+    @Test(expected = InvalidOSMDataException.class) // minification empty string osmData
```
Unimportant comment. You need to remove it.

VladimirBalun commented on ServerSide/resourceserver/src/test/java/ru/servers/resourceserver/OSMMinifierTest.java:
```diff
@@ -0,0 +1,107 @@
+                "<?xml version=\"1.0\" encoding=\"UTF-8\"?> " +
+                        "<osm attribution=\"http://www.openstreetmap.org/copyright\" copyright=\"OpenStreetMap and contributors\" generator=\"CGImap 0.6.1 (11633 thorn-03.openstreetmap.org)\" license=\"http://opendatacommons.org/licenses/odbl/1-0/\" version=\"0.6\">" +
+                        "<way id=\"580429971\" visible=\"true\" version=\"1\" changeset=\"58135270\" timestamp=\"2018-04-16T11:43:32Z\" user=\"Alex_Enot\" uid=\"5757927\"></way>";
+        final String minifiedData = "";
```
I did not know this design...
'minifiedData' must not be empty, method should throw an exception.

VladimirBalun commented on ServerSide/resourceserver/src/test/java/ru/servers/resourceserver/OSMMinifierTest.java:
```diff
@@ -0,0 +1,107 @@
+        final String minifiedData = "";
+
+        String resultMinification = osmMinifier.minify(sourceData);
+        Assert.assertEquals(minifiedData, resultMinification);
```
I did not know this design...
'minifiedData' must not be empty, method should throw an exception.

VladimirBalun commented on ServerSide/resourceserver/src/test/java/ru/servers/resourceserver/OSMMinifierTest.java:
```diff
@@ -0,0 +1,107 @@
+        String resultMinification = osmMinifier.minify(sourceData);
+        Assert.assertEquals(minifiedData, resultMinification);
+    }
+
```
The next two methods are equals. You should use one of them.

VladimirBalun commented:
You also need fix these warnings with codacy: https://app.codacy.com/app/VladimirBalun/RacingWorld/pullRequest?prid=3519170 , and resolve merge conflicts in the last commit.

Smipe-a committed:
refactor(server side): deleted extra comments, changed exception class

Was deleted extra comments, changed exception class, deleted empty lines and superfluous data normalization method.

Smipe-a committed:
refactor(server side): changed returning result method normalizationData

Was changed returning result method normalizationData.

Smipe-a committed:
feat(server side): changed method normalizationData

Was changed method normalizationData.

VladimirBalun reviewed (commented):
Good work.
